doc/source/whatsnew/v0.20.0.txt

```diff
@@ -79,3 +79,5 @@ Performance Improvements
 
 Bug Fixes
 ~~~~~~~~~
+
+- Bug in Numpy ``astype()`` caused `inf` values to be falsely converted to integers. Affected ``astype()`` in Series and DataFrames (:issue:`14265`)
```

Comment:
I wouldn't call this a "bug in numpy". It is defined behaviour of numpy, which we just don't want to expose to our users and therefore provide an error (as we already do with NaN)